src/main/java/com/petros/bibernate/util/EntityUtil.java

```diff
@@ -122,20 +118,16 @@ public static List<String> getInsertableColumns(Class<?> entityClass) {
 
     /**
      * Retrieves the list of entity values for a given entity.
+     * Fields marked with {@link OneToMany} annotation are ignored
      *
      * @param entity Bibernate entity
      * @return the list of entity values
      */
-    public static List<Object> getEntityFields(Object entity) {
+    public static List<Object> getEntityFieldsForSnapshot(Object entity) {
         return Arrays.stream(entity.getClass().getDeclaredFields())
                 .peek(field -> field.setAccessible(true))
-                .map(field -> {
-                    try {
-                        return field.get(entity);
-                    } catch (IllegalAccessException e) {
-                        throw new BibernateException(e);
-                    }
-                })
+                .filter(field -> !isEntityCollectionField(field))
```

Comment:
We should also exclude @ManyToOne (and @OneToOne when it becomes available). For instance, we won't be able to create a snapshot for a Person.